### PR TITLE
feat: subtree get/insert

### PR DESCRIPTION
**Pull request by fominok**
This is an attempt to implement simple insert and get operations for GroveDb (inside subtrees without exposing to public API yet) to store binary data

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -23,24 +30,30 @@ pub struct GroveDb {
 impl GroveDb {
     pub fn new() -> Result<Self, Error> {
         let subtrees = vec![
-            Merk::open("./test1.db")?,
-            Merk::open("./test2.db")?,
-            Merk::open("./test3.db")?,
-            Merk::open("./test4.db")?,
-            Merk::open("./test5.db")?,
-            Merk::open("./test6.db")?,
+            Merk::open(Path::new(SUBTREES_DIR).join("test1.db"))?,
```
Comment (shumkov) [inline src/lib.rs:33]: All trees need to be stored in one database

Comment (fominok) [reply src/lib.rs:33]: Yeah, it will be a scope of a next task

File: src/subtree.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                .ok_or(Error::InvalidPath("key not found in Merk"))?
+                .as_slice(),
+        )?;
+        element.follow_reference(&merk)
+    }
+
+    pub fn insert(&self, merk: &mut Merk, path: &[&[u8]], key: &[u8]) -> Result<(), Error> {
+        // check if a tree was inserted by the path
+        if let Some((tree_key, tree_path)) = path.split_last() {
+            Element::get(merk, tree_path, tree_key)?.is_tree()?;
```
Comment (QuantumExplorer) [inline src/subtree.rs:81]: Why use is_tree? Shouldn't this fail no matter what if an element exists already?

**Comment by QuantumExplorer** [comment]
Very good initial work guys!

File: src/subtree.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    /// Recursively follow `Element::Reference`
+    fn follow_reference(self, merk: &Merk) -> Result<Element, Error> {
+        if let Element::Reference(reference_merk_key) = self {
+            let element = Element::decode(
+                merk.get(reference_merk_key.as_slice())?
+                    .ok_or(Error::InvalidPath("key not found in Merk"))?
+                    .as_slice(),
+            )?;
+            element.follow_reference(merk)
```
[the remark this answers is not in the document]
Comment (fominok) [reply src/subtree.rs:39]: Good catch, thank you

File: src/subtree.rs (hunk 2 above)
Comment (fominok) [reply src/subtree.rs:81]: It checks whether there is a tree at path, it uses `split_last` to get path part and last path element to reuse our `get` method which expects a key, so in this case the key will be last element of `path`

Tldr: it checks the presence of a tree to insert key into

Comment (QuantumExplorer) [reply src/subtree.rs:81]: Sorry I misunderstood the code.

File: src/lib.rs (hunk 1 above)
Comment (fominok) [reply src/lib.rs]: Actually, done in this PR, please take a look

**Review by QuantumExplorer — APPROVED** [review]
